client/starwhale/api/_impl/track/collector.py

```diff
@@ -0,0 +1,238 @@
+            git_info = {}
+
+        return {
+            "code": {
+                "git": git_info,
+                "workdir": str(self.workdir),
+            }
+        }
+
+    def _inspect_python_run(self) -> t.Dict[str, t.Any]:
+        # TODO: add starwhale runtime, pip dependencies(pip requirements.txt and conda export.yaml) info
+        return {
+            "python": {
+                "version": platform.python_version(),
+                "bin_path": sys.executable,
+                "env_mode": guess_current_py_env(),
+            }
+        }
+
+    def _inspect_system_specs(self) -> t.Dict[str, t.Any]:
+        # TODO: add more system specs: gpu, network, ip...
+        return {
+            "system": {
+                "hostname": platform.node(),
+                "platform": sys.platform,
+                "os": platform.platform(),
+                "arch": platform.machine(),
+            },
+            "hardware": {
+                "cpu": {
+                    "cores": psutil.cpu_count(logical=False),
+                    "cores_logical": psutil.cpu_count(logical=True),
+                },
+                "memory": {"total_bytes": psutil.virtual_memory().total},
+                "disk": {
+                    "total_bytes": psutil.disk_usage("/").total,
+                    "used_bytes": psutil.disk_usage("/").used,
+                },
+            },
+        }
+
+    def _inspect_process_specs(self) -> t.Dict[str, t.Any]:
+        # TODO: get process more info
+        return {
+            "process": {
+                "pid": self._pid,
+                "cwd": str(self.workdir),
+                "cmdline": " ".join(psutil.Process(self._pid).cmdline()),
+            },
+        }
+
+    def _inspect_environments(self) -> t.Dict[str, t.Any]:
+        return {
+            "environments": dict(os.environ),
+        }
+
+    def _inspect_metrics(self) -> t.Dict[str, t.Any]:
+        _process = psutil.Process(self._pid)
+        # TODO: support more system and process metrics: network, io, memory
+        system_cpu_percent = [float(v) for v in psutil.cpu_percent(percpu=True)]  # type: ignore
+        cpu_cores = float(len(system_cpu_percent))
+        return {
+            "system": {
+                "cpu": {
+                    "usage_percent": sum(system_cpu_percent) / cpu_cores,
+                },
+            },
+            "process": {
+                "cpu": {
+                    "usage_percent": _process.cpu_percent() / cpu_cores,
+                },
+                "num_threads": _process.num_threads(),
+            },
+        }
+
+    def _report_metrics(self) -> None:
+        data = {}
+        for k, v in self._staging_metrics.items():
+            if not v:
+                continue
+
+            data[f"{k}/last"] = float(v[-1])
+            data[f"{k}/avg"] = round(sum(v) / len(v), 3)
+
+        if not data:
+            return
+
+        self.tracker._log_metrics(
```

Comment:
should the func named to 'log_metrics'?

Reply:
`_log_metrics` is an internal function. The user should use `track.metrics()` function.

`track._log_metrics` -> `track.metrics`
`track._log_params` -> `track.params`
`track._log_artifacts` -> `track.artifacts`